app/src/main/java/com/example/weatherapp/ForecastRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.weatherapp
+
+import android.content.Context
+import android.content.res.Resources
+import android.os.Build
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.annotation.RequiresApi
+import androidx.recyclerview.widget.RecyclerView
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+
+class ForecastRecyclerViewAdapter(private val data: List<DayForecast>, private val context: Context)
```

Comment:
You should not pass the activity context to the Adapter. Instead you can get access to the context through the itemView in the ViewHolder. For example: `itemView.context.getString()`.